sigopt/xgboost/experiment.py

```diff
@@ -36,17 +47,52 @@ def parse_and_create_experiment(self):
         else:
           metric['name'] = DEFAULT_EVALS_NAME + '-' + metric['name']
 
+  def parse_and_create_parameters(self):
+    # Set defaults as needed
+    if 'budget' not in self.experiment_config_parsed:
+      self.experiment_config_parsed['budget'] = DEFAULT_BO_ITERATIONS
+    if 'parallel_bandwidth' not in self.experiment_config_parsed:
+      self.experiment_config_parsed['parallel_bandwidth'] = 1
+    if 'type' not in self.experiment_config_parsed:
+      self.experiment_config_parsed['type'] = 'offline'
+
+    # Parse params
+    if 'parameters' not in self.experiment_config_parsed:
+      self.experiment_config_parsed['parameters'] = [
+        SEARCH_BOUNDS[SEARCH_PARAMS.index(param_name)] for param_name in DEFAULT_SEARCH_PARAMS
+      ]
+    else:
+      for param in self.experiment_config_parsed['parameters']:
+        if 'bounds' not in param:
+          param_name = param['name']
+          assert param_name in SEARCH_PARAMS, f'We do not support autoselection of bounds for {param_name}'
+          search_bound = SEARCH_BOUNDS[SEARCH_PARAMS.index(param_name)]
+          param['bounds'] = search_bound['bounds']
+          param['type'] = search_bound['type']
+          if 'transformation' in search_bound:
+            param['transformation'] = search_bound['transformation']
+
+  def parse_and_create_experiment_config(self):
+    self.experiment_config_parsed = copy.deepcopy(self.experiment_config)
```

Comment:
It is better in `__init__()`, `self.experiment_config = copy.deepcopy(experiment_config)`.

Reply:
agreed. Do we need to preserve `self.experiment_config`?